### PR TITLE
test: initial tracing-cov-mark crate implementation

### PR DESCRIPTION
**Pull request by CBenoit**
@irvingoujAtDevolution this is what I mentioned you earlier this week.
We can assert that some code path has been exercised using this tracing subscriber and helper.

### CODE REVIEW

File: crates/network-scanner-net/src/runtime.rs [hunk 1]
```diff
@@ -29,14 +29,17 @@ pub struct Socket2Runtime {
 
 impl Drop for Socket2Runtime {
     fn drop(&mut self) {
-        tracing::debug!("dropping runtime");
+        tracing::trace!(covmark = "socket2_runtime_drop");
```
Comment (CBenoit) [inline crates/network-scanner-net/src/runtime.rs:32]: First, you emit a tracing event with a "covmark" field…

File: crates/network-scanner-net/src/tests.rs [hunk 2]
```diff
@@ -208,21 +212,25 @@ pub async fn drop_runtime() -> anyhow::Result<()> {
...
         }
-        tracing::info!("runtime arc count: {}", Arc::strong_count(&async_runtime));
 
-        assert!(Arc::strong_count(&async_runtime) == 1);
+        tracing::info!("runtime arc count: {}", Arc::strong_count(&runtime));
+        assert!(Arc::strong_count(&runtime) == 1);
     }
+
     tracing::info!("runtime should be dropped here");
+    cov.assert_mark("socket2_runtime_drop");
```
Comment (CBenoit) [inline crates/network-scanner-net/src/tests.rs:229]: …and then you can assert the event has been fired in the test case.

**Comment by CBenoit** [comment]
Rebased on top of master

**Review by irvingoujAtDevolution — APPROVED** [review]

**Comment by irvingoujAtDevolution** [comment]
Thank you, learned something new!

**Pull request merged**